# Let us import the necessary libraries
#### We would be needing pandas library to read the csv file, numpy libraries to work on calculating parameters like msre, rss etc. we would need seaborn and matplot lib to visually learn if there is a linear relationship between parameters. We would be using sklearn libraries to build a linear model, use regularlization like lasso and ridge.


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime

import warnings
warnings.filterwarnings("ignore")

# Data Understanding and Exploration

In [2]:
data = pd.read_csv("train.csv")
pd.options.display.max_columns=100
data.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities LotConfig LandSlope Neighborhood Condition1  \
0         Lvl    AllPub    Inside       Gtl      CollgCr       Norm   
1         Lvl    AllPub       FR2       Gtl      Veenker      Feedr   
2         Lvl    AllPub    Inside       Gtl      CollgCr       Norm   
3         Lvl    AllPub    Corner       Gtl      Crawfor       Norm   
4         Lvl    AllPub       FR2       Gtl      NoRidge       Norm   

  Condition2 BldgType HouseStyle  OverallQual  OverallCond  YearBuilt  \
0       Norm     1Fam     2Story            7            5       2003   
1       Norm     1Fam     1Story            6            8       1976   
2       Norm     1Fam     2Story            7            5       2001   
3       Norm     1Fam     2Story            7            5       1915   
4       Norm     1Fam     2Story            8            5       2000   

   YearRemodAdd RoofStyle RoofMatl Exterior1st Exterior2nd MasVnrType  \
0          2003     Gable  CompShg     VinylSd     VinylSd    BrkFace   
1          1976     Gable  CompShg     MetalSd     MetalSd        NaN   
2          2002     Gable  CompShg     VinylSd     VinylSd    BrkFace   
3          1970     Gable  CompShg     Wd Sdng     Wd Shng        NaN   
4          2000     Gable  CompShg     VinylSd     VinylSd    BrkFace   

   MasVnrArea ExterQual ExterCond Foundation BsmtQual BsmtCond BsmtExposure  \
0       196.0        Gd        TA      PConc       Gd       TA           No   
1         0.0        TA        TA     CBlock       Gd       TA           Gd   
2       162.0        Gd        TA      PConc       Gd       TA           Mn   
3         0.0        TA        TA     BrkTil       TA       Gd           No   
4       350.0        Gd        TA      PConc       Gd       TA           Av   

  BsmtFinType1  BsmtFinSF1 BsmtFinType2  BsmtFinSF2  BsmtUnfSF  TotalBsmtSF  \
0          GLQ         706          Unf           0        150          856   
1          ALQ         978          Unf           0        284         1262   
2          GLQ         486          Unf           0        434          920   
3          ALQ         216          Unf           0        540          756   
4          GLQ         655          Unf           0        490         1145   

  Heating HeatingQC CentralAir Electrical  1stFlrSF  2ndFlrSF  LowQualFinSF  \
0    GasA        Ex          Y      SBrkr       856       854             0   
1    GasA        Ex          Y      SBrkr      1262         0             0   
2    GasA        Ex          Y      SBrkr       920       866             0   
3    GasA        Gd          Y      SBrkr       961       756             0   
4    GasA        Ex          Y      SBrkr      1145      1053             0   

   GrLivArea  BsmtFullBath  BsmtHalfBath  FullBath  HalfBath  BedroomAbvGr  \
0       1710             1             0         2         1             3   
1       1262             0             1         2         0             3   
2       1786             1             0         2         1             3   
3       1717             1             0         1         0             3   
4       2198             1             0         2         1             4   

   KitchenAbvGr KitchenQual  TotRmsAbvGrd Functional  Fireplaces FireplaceQu  \
0             1          Gd             8        Typ           0         NaN   
1             1          TA             6        Typ           1          TA   
2             1          Gd             6        Typ           1          TA   
3             1          Gd            

In [3]:
data.shape

(1460, 81)

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

We see that there are few columns having null values.
For example Alley, PoolQC, Fence, MiscFeatures, these columns have got very few not null values out of total 1460. Hence these columns should not contribute in building the model.
We should be dropping Id column also as it is not significant in model building

In [5]:
columns_to_drop = ["Alley","PoolQC","Fence","MiscFeature","Id"]
data = data.drop(columns=columns_to_drop)

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 76 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   MSSubClass     1460 non-null   int64  
 1   MSZoning       1460 non-null   object 
 2   LotFrontage    1201 non-null   float64
 3   LotArea        1460 non-null   int64  
 4   Street         1460 non-null   object 
 5   LotShape       1460 non-null   object 
 6   LandContour    1460 non-null   object 
 7   Utilities      1460 non-null   object 
 8   LotConfig      1460 non-null   object 
 9   LandSlope      1460 non-null   object 
 10  Neighborhood   1460 non-null   object 
 11  Condition1     1460 non-null   object 
 12  Condition2     1460 non-null   object 
 13  BldgType       1460 non-null   object 
 14  HouseStyle     1460 non-null   object 
 15  OverallQual    1460 non-null   int64  
 16  OverallCond    1460 non-null   int64  
 17  YearBuilt      1460 non-null   int64  
 18  YearRemo

In [7]:
data.head(10)

MSSubClass MSZoning  LotFrontage  LotArea Street LotShape LandContour  \
0          60       RL         65.0     8450   Pave      Reg         Lvl   
1          20       RL         80.0     9600   Pave      Reg         Lvl   
2          60       RL         68.0    11250   Pave      IR1         Lvl   
3          70       RL         60.0     9550   Pave      IR1         Lvl   
4          60       RL         84.0    14260   Pave      IR1         Lvl   
5          50       RL         85.0    14115   Pave      IR1         Lvl   
6          20       RL         75.0    10084   Pave      Reg         Lvl   
7          60       RL          NaN    10382   Pave      IR1         Lvl   
8          50       RM         51.0     6120   Pave      Reg         Lvl   
9         190       RL         50.0     7420   Pave      Reg         Lvl   

  Utilities LotConfig LandSlope Neighborhood Condition1 Condition2 BldgType  \
0    AllPub    Inside       Gtl      CollgCr       Norm       Norm     1Fam   
1    AllPub       FR2       Gtl      Veenker      Feedr       Norm     1Fam   
2    AllPub    Inside       Gtl      CollgCr       Norm       Norm     1Fam   
3    AllPub    Corner       Gtl      Crawfor       Norm       Norm     1Fam   
4    AllPub       FR2       Gtl      NoRidge       Norm       Norm     1Fam   
5    AllPub    Inside       Gtl      Mitchel       Norm       Norm     1Fam   
6    AllPub    Inside       Gtl      Somerst       Norm       Norm     1Fam   
7    AllPub    Corner       Gtl       NWAmes       PosN       Norm     1Fam   
8    AllPub    Inside       Gtl      OldTown     Artery       Norm     1Fam   
9    AllPub    Corner       Gtl      BrkSide     Artery     Artery   2fmCon   

  HouseStyle  OverallQual  OverallCond  YearBuilt  YearRemodAdd RoofStyle  \
0     2Story            7            5       2003          2003     Gable   
1     1Story            6            8       1976          1976     Gable   
2     2Story            7            5       2001          2002     Gable   
3     2Story            7            5       1915          1970     Gable   
4     2Story            8            5       2000          2000     Gable   
5     1.5Fin            5            5       1993          1995     Gable   
6     1Story            8            5       2004          2005     Gable   
7     2Story            7            6       1973          1973     Gable   
8     1.5Fin            7            5       1931          1950     Gable   
9     1.5Unf            5            6       1939          1950     Gable   

  RoofMatl Exterior1st Exterior2nd MasVnrType  MasVnrArea ExterQual ExterCond  \
0  CompShg     VinylSd     VinylSd    BrkFace       196.0        Gd        TA   
1  CompShg     MetalSd     MetalSd        NaN         0.0        TA        TA   
2  CompShg     VinylSd     VinylSd    BrkFace       162.0        Gd        TA   
3  CompShg     Wd Sdng     Wd Shng        NaN         0.0        TA        TA   
4  CompShg     VinylSd     VinylSd    BrkFace       350.0        Gd        TA   
5  CompShg     VinylSd     VinylSd        NaN         0.0        TA        TA   
6  CompShg     VinylSd     VinylSd      Stone       186.0        Gd        TA   
7  CompShg     HdBoard     HdBoard      Stone       240.0        TA        TA   
8  CompShg     BrkFace     Wd Shng        NaN         0.0        TA        TA   
9  CompShg     MetalSd     MetalSd        NaN         0.0        TA        TA   

  Foundation BsmtQual BsmtCond BsmtExposure BsmtFinType1  BsmtFinSF1  \
0      PConc       Gd       TA           No          GLQ         706   
1     CBlock       Gd       TA           Gd          ALQ         978   
2      PConc       Gd       TA           Mn          GLQ         486   
3     BrkTil       TA       Gd           No          ALQ         216   
4      PConc       Gd       TA           Av          GLQ         655   
5       Wood       Gd       TA           No          GLQ         732   
6      PConc       Ex       TA           Av         

##### There are still few columns like LotFrontage which is having null values. Let's try to replace those null values 

In [8]:
data["LotFrontage"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 1460 entries, 0 to 1459
Series name: LotFrontage
Non-Null Count  Dtype  
--------------  -----  
1201 non-null   float64
dtypes: float64(1)
memory usage: 11.5 KB


###### This is a numeric column hence we can replace the null values with zero

In [9]:
data["LotFrontage"] = data["LotFrontage"].fillna(0)
data["LotFrontage"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 1460 entries, 0 to 1459
Series name: LotFrontage
Non-Null Count  Dtype  
--------------  -----  
1460 non-null   float64
dtypes: float64(1)
memory usage: 11.5 KB


##### Based on our understanding of data dictionary, there are few variables which logically does not mak sense in building the model 
For example we see the column Neighborhood which is the physical location of the property. However we see there are two more columns Condition1 & Condition2 which also describe Proximity to various conditions.

###### Hence we would be picking handful of columns for our analysis from the entire dataset

In [10]:
variables_for_model_building = ["LotArea","MSSubClass","MSZoning","Utilities","Neighborhood","BldgType","TotalBsmtSF",
                                "Heating","CentralAir","1stFlrSF","2ndFlrSF","BedroomAbvGr","KitchenAbvGr",
                                "GarageArea","PoolArea","YearBuilt","YearRemodAdd","Electrical","1stFlrSF","2ndFlrSF",
                                "GarageType","SalePrice"]


In [11]:
housing_dataset = data[variables_for_model_building]
housing_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 22 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   LotArea       1460 non-null   int64 
 1   MSSubClass    1460 non-null   int64 
 2   MSZoning      1460 non-null   object
 3   Utilities     1460 non-null   object
 4   Neighborhood  1460 non-null   object
 5   BldgType      1460 non-null   object
 6   TotalBsmtSF   1460 non-null   int64 
 7   Heating       1460 non-null   object
 8   CentralAir    1460 non-null   object
 9   1stFlrSF      1460 non-null   int64 
 10  2ndFlrSF      1460 non-null   int64 
 11  BedroomAbvGr  1460 non-null   int64 
 12  KitchenAbvGr  1460 non-null   int64 
 13  GarageArea    1460 non-null   int64 
 14  PoolArea      1460 non-null   int64 
 15  YearBuilt     1460 non-null   int64 
 16  YearRemodAdd  1460 non-null   int64 
 17  Electrical    1459 non-null   object
 18  1stFlrSF      1460 non-null   int64 
 19  2ndFlr

In [12]:
housing_dataset.head(5)

LotArea  MSSubClass MSZoning Utilities Neighborhood BldgType  TotalBsmtSF  \
0     8450          60       RL    AllPub      CollgCr     1Fam          856   
1     9600          20       RL    AllPub      Veenker     1Fam         1262   
2    11250          60       RL    AllPub      CollgCr     1Fam          920   
3     9550          70       RL    AllPub      Crawfor     1Fam          756   
4    14260          60       RL    AllPub      NoRidge     1Fam         1145   

  Heating CentralAir  1stFlrSF  2ndFlrSF  BedroomAbvGr  KitchenAbvGr  \
0    GasA          Y       856       854             3             1   
1    GasA          Y      1262         0             3             1   
2    GasA          Y       920       866             3             1   
3    GasA          Y       961       756             3             1   
4    GasA          Y      1145      1053             4             1   

   GarageArea  PoolArea  YearBuilt  YearRemodAdd Electrical  1stFlrSF  \
0         548         0       2003          2003      SBrkr       856   
1         460         0       1976          1976      SBrkr      1262   
2         608         0       2001          2002      SBrkr       920   
3         642         0       1915          1970      SBrkr       961   
4         836         0       2000          2000      SBrkr      1145   

   2ndFlrSF GarageType  SalePrice  
0       854     Attchd     208500  
1         0     Attchd     181500  
2       866     Attchd     223500  
3       756     Detchd     140000  
4      1053     Attchd     250000

##### Of the 21 columns we have selected, we see some are of categorical type, let us convert them to dummy variables

In [13]:
categorical_variables = ["Neighborhood","MSSubClass","MSZoning","Utilities","BldgType","Heating","Electrical","GarageType"]

###### Let us loop through all categorical variables and create dummy variables dataframe &  concat the dataframe of dummy variables with main dataset

In [14]:
for categorical_var in categorical_variables:
    print("Creating dummy variabels for categorical variable {}".format(categorical_var))
    dummy_var_df = pd.get_dummies(housing_dataset[categorical_var],prefix=categorical_var+"_",dtype=int,drop_first=True)
    housing_dataset = pd.concat([housing_dataset,dummy_var_df],axis=1)

Creating dummy variabels for categorical variable Neighborhood
Creating dummy variabels for categorical variable MSSubClass
Creating dummy variabels for categorical variable MSZoning
Creating dummy variabels for categorical variable Utilities
Creating dummy variabels for categorical variable BldgType
Creating dummy variabels for categorical variable Heating
Creating dummy variabels for categorical variable Electrical
Creating dummy variabels for categorical variable GarageType


In [15]:
housing_dataset.drop(columns=categorical_variables,inplace=True)
housing_dataset.shape

(1460, 75)

In [16]:
housing_dataset.head(10)

LotArea  TotalBsmtSF CentralAir  1stFlrSF  2ndFlrSF  BedroomAbvGr  \
0     8450          856          Y       856       854             3   
1     9600         1262          Y      1262         0             3   
2    11250          920          Y       920       866             3   
3     9550          756          Y       961       756             3   
4    14260         1145          Y      1145      1053             4   
5    14115          796          Y       796       566             1   
6    10084         1686          Y      1694         0             3   
7    10382         1107          Y      1107       983             3   
8     6120          952          Y      1022       752             2   
9     7420          991          Y      1077         0             2   

   KitchenAbvGr  GarageArea  PoolArea  YearBuilt  YearRemodAdd  1stFlrSF  \
0             1         548         0       2003          2003       856   
1             1         460         0       1976          1976      1262   
2             1         608         0       2001          2002       920   
3             1         642         0       1915          1970       961   
4             1         836         0       2000          2000      1145   
5             1         480         0       1993          1995       796   
6             1         636         0       2004          2005      1694   
7             1         484         0       1973          1973      1107   
8             2         468         0       1931          1950      1022   
9             2         205         0       1939          1950      1077   

   2ndFlrSF  SalePrice  Neighborhood__Blueste  Neighborhood__BrDale  \
0       854     208500                      0                     0   
1         0     181500                      0                     0   
2       866     223500                      0                     0   
3       756     140000                      0                     0   
4      1053     250000                      0                     0   
5       566     143000                      0                     0   
6         0     307000                      0                     0   
7       983     200000                      0                     0   
8       752     129900                      0                     0   
9         0     118000                      0                     0   

   Neighborhood__BrkSide  Neighborhood__ClearCr  Neighborhood__CollgCr  \
0                      0                      0                      1   
1                      0                      0                      0   
2                      0                      0                      1   
3                      0                      0                      0   
4                      0                      0                      0   
5                      0                      0                      0   
6                      0                      0                      0   
7                      0                      0                      0   
8                      0                      0                      0   
9                      1                      0                      0   

   Neighborhood__Crawfor  Neighborhood__Edwards  Neighborhood__Gilbert  \
0                      0                      0                      0   
1                      0                      0                      0   
2                      0                      0                      0   
3                      1                      0                      0   
4                      0                      0                      0   
5                      0                      0                      0   
6                      0                      0                      0   
7                      0                      0                      0   
8                      0                      0                      0   
9                      0            

###### CentralAir is a binary categorical variable, let's replace it with binary values

In [17]:
housing_dataset["CentralAir"] = housing_dataset["CentralAir"].map({'Y':1,'N':0})
housing_dataset.head()

LotArea  TotalBsmtSF  CentralAir  1stFlrSF  2ndFlrSF  BedroomAbvGr  \
0     8450          856           1       856       854             3   
1     9600         1262           1      1262         0             3   
2    11250          920           1       920       866             3   
3     9550          756           1       961       756             3   
4    14260         1145           1      1145      1053             4   

   KitchenAbvGr  GarageArea  PoolArea  YearBuilt  YearRemodAdd  1stFlrSF  \
0             1         548         0       2003          2003       856   
1             1         460         0       1976          1976      1262   
2             1         608         0       2001          2002       920   
3             1         642         0       1915          1970       961   
4             1         836         0       2000          2000      1145   

   2ndFlrSF  SalePrice  Neighborhood__Blueste  Neighborhood__BrDale  \
0       854     208500                      0                     0   
1         0     181500                      0                     0   
2       866     223500                      0                     0   
3       756     140000                      0                     0   
4      1053     250000                      0                     0   

   Neighborhood__BrkSide  Neighborhood__ClearCr  Neighborhood__CollgCr  \
0                      0                      0                      1   
1                      0                      0                      0   
2                      0                      0                      1   
3                      0                      0                      0   
4                      0                      0                      0   

   Neighborhood__Crawfor  Neighborhood__Edwards  Neighborhood__Gilbert  \
0                      0                      0                      0   
1                      0                      0                      0   
2                      0                      0                      0   
3                      1                      0                      0   
4                      0                      0                      0   

   Neighborhood__IDOTRR  Neighborhood__MeadowV  Neighborhood__Mitchel  \
0                     0                      0                      0   
1                     0                      0                      0   
2                     0                      0                      0   
3                     0                      0                      0   
4                     0                      0                      0   

   Neighborhood__NAmes  Neighborhood__NPkVill  Neighborhood__NWAmes  \
0                    0                      0                     0   
1                    0                      0                     0   
2                    0                      0                     0   
3                    0                      0                     0   
4                    0                      0                     0   

   Neighborhood__NoRidge  Neighborhood__NridgHt  Neighborhood__OldTown  \
0                      0                      0                      0   
1                      0                      0                      0   
2                      0                      0                      0   
3                      0                      0                      0   
4                      1                      0                      0   

   Neighborhood__SWISU  Neighborhood__Sawyer  Neighborhood__SawyerW  \
0                    0                     0                      0   
1                    0                     0                      0   
2                    0                     0                      0   
3                    0                     0                      0   
4                    0                     0                      0   

   Neighborhood__Somerst  Neighborhood__StoneBr  Nei

##### Columns YearBuilt and YearRemodAdd will have more impact on sale of the property as the one that was build 10 years back will be sold at a higher price than the one that was built 30 years back. 

Let us convert this column into a numeric column ageOfProperty describing the age of the property. If property that was built in 2014 will have age of 10 years, hence the column ageOfProperty will have numeric value 10 

In [18]:
current_year = datetime.now().year
housing_dataset["ageOfProperty"] =  current_year - housing_dataset["YearBuilt"]

In [19]:
housing_dataset.head()

LotArea  TotalBsmtSF  CentralAir  1stFlrSF  2ndFlrSF  BedroomAbvGr  \
0     8450          856           1       856       854             3   
1     9600         1262           1      1262         0             3   
2    11250          920           1       920       866             3   
3     9550          756           1       961       756             3   
4    14260         1145           1      1145      1053             4   

   KitchenAbvGr  GarageArea  PoolArea  YearBuilt  YearRemodAdd  1stFlrSF  \
0             1         548         0       2003          2003       856   
1             1         460         0       1976          1976      1262   
2             1         608         0       2001          2002       920   
3             1         642         0       1915          1970       961   
4             1         836         0       2000          2000      1145   

   2ndFlrSF  SalePrice  Neighborhood__Blueste  Neighborhood__BrDale  \
0       854     208500                      0                     0   
1         0     181500                      0                     0   
2       866     223500                      0                     0   
3       756     140000                      0                     0   
4      1053     250000                      0                     0   

   Neighborhood__BrkSide  Neighborhood__ClearCr  Neighborhood__CollgCr  \
0                      0                      0                      1   
1                      0                      0                      0   
2                      0                      0                      1   
3                      0                      0                      0   
4                      0                      0                      0   

   Neighborhood__Crawfor  Neighborhood__Edwards  Neighborhood__Gilbert  \
0                      0                      0                      0   
1                      0                      0                      0   
2                      0                      0                      0   
3                      1                      0                      0   
4                      0                      0                      0   

   Neighborhood__IDOTRR  Neighborhood__MeadowV  Neighborhood__Mitchel  \
0                     0                      0                      0   
1                     0                      0                      0   
2                     0                      0                      0   
3                     0                      0                      0   
4                     0                      0                      0   

   Neighborhood__NAmes  Neighborhood__NPkVill  Neighborhood__NWAmes  \
0                    0                      0                     0   
1                    0                      0                     0   
2                    0                      0                     0   
3                    0                      0                     0   
4                    0                      0                     0   

   Neighborhood__NoRidge  Neighborhood__NridgHt  Neighborhood__OldTown  \
0                      0                      0                      0   
1                      0                      0                      0   
2                      0                      0                      0   
3                      0                      0                      0   
4                      1                      0                      0   

   Neighborhood__SWISU  Neighborhood__Sawyer  Neighborhood__SawyerW  \
0                    0                     0                      0   
1                    0                     0                      0   
2                    0                     0                      0   
3                    0                     0                      0   
4                    0                     0                      0   

   Neighborhood__Somerst  Neighborhood__StoneBr  Nei

###### Simillary we can create another column ageSinceRemodel which would be current year subtracted by YearRemodAdd that will describe age of the property since it has been remodeled

In [20]:
housing_dataset["ageSinceRemodel"] =  current_year - housing_dataset["YearRemodAdd"]

In [21]:
housing_dataset.head()

LotArea  TotalBsmtSF  CentralAir  1stFlrSF  2ndFlrSF  BedroomAbvGr  \
0     8450          856           1       856       854             3   
1     9600         1262           1      1262         0             3   
2    11250          920           1       920       866             3   
3     9550          756           1       961       756             3   
4    14260         1145           1      1145      1053             4   

   KitchenAbvGr  GarageArea  PoolArea  YearBuilt  YearRemodAdd  1stFlrSF  \
0             1         548         0       2003          2003       856   
1             1         460         0       1976          1976      1262   
2             1         608         0       2001          2002       920   
3             1         642         0       1915          1970       961   
4             1         836         0       2000          2000      1145   

   2ndFlrSF  SalePrice  Neighborhood__Blueste  Neighborhood__BrDale  \
0       854     208500                      0                     0   
1         0     181500                      0                     0   
2       866     223500                      0                     0   
3       756     140000                      0                     0   
4      1053     250000                      0                     0   

   Neighborhood__BrkSide  Neighborhood__ClearCr  Neighborhood__CollgCr  \
0                      0                      0                      1   
1                      0                      0                      0   
2                      0                      0                      1   
3                      0                      0                      0   
4                      0                      0                      0   

   Neighborhood__Crawfor  Neighborhood__Edwards  Neighborhood__Gilbert  \
0                      0                      0                      0   
1                      0                      0                      0   
2                      0                      0                      0   
3                      1                      0                      0   
4                      0                      0                      0   

   Neighborhood__IDOTRR  Neighborhood__MeadowV  Neighborhood__Mitchel  \
0                     0                      0                      0   
1                     0                      0                      0   
2                     0                      0                      0   
3                     0                      0                      0   
4                     0                      0                      0   

   Neighborhood__NAmes  Neighborhood__NPkVill  Neighborhood__NWAmes  \
0                    0                      0                     0   
1                    0                      0                     0   
2                    0                      0                     0   
3                    0                      0                     0   
4                    0                      0                     0   

   Neighborhood__NoRidge  Neighborhood__NridgHt  Neighborhood__OldTown  \
0                      0                      0                      0   
1                      0                      0                      0   
2                      0                      0                      0   
3                      0                      0                      0   
4                      1                      0                      0   

   Neighborhood__SWISU  Neighborhood__Sawyer  Neighborhood__SawyerW  \
0                    0                     0                      0   
1                    0                     0                      0   
2                    0                     0                      0   
3                    0                     0                      0   
4                    0                     0                      0   

   Neighborhood__Somerst  Neighborhood__StoneBr  Nei

Let us drop the variables YearBuilt & YearRemodAdd from the dataset

In [22]:
housing_dataset = housing_dataset.drop(columns=["YearBuilt","YearRemodAdd"])
housing_dataset.head()

LotArea  TotalBsmtSF  CentralAir  1stFlrSF  2ndFlrSF  BedroomAbvGr  \
0     8450          856           1       856       854             3   
1     9600         1262           1      1262         0             3   
2    11250          920           1       920       866             3   
3     9550          756           1       961       756             3   
4    14260         1145           1      1145      1053             4   

   KitchenAbvGr  GarageArea  PoolArea  1stFlrSF  2ndFlrSF  SalePrice  \
0             1         548         0       856       854     208500   
1             1         460         0      1262         0     181500   
2             1         608         0       920       866     223500   
3             1         642         0       961       756     140000   
4             1         836         0      1145      1053     250000   

   Neighborhood__Blueste  Neighborhood__BrDale  Neighborhood__BrkSide  \
0                      0                     0                      0   
1                      0                     0                      0   
2                      0                     0                      0   
3                      0                     0                      0   
4                      0                     0                      0   

   Neighborhood__ClearCr  Neighborhood__CollgCr  Neighborhood__Crawfor  \
0                      0                      1                      0   
1                      0                      0                      0   
2                      0                      1                      0   
3                      0                      0                      1   
4                      0                      0                      0   

   Neighborhood__Edwards  Neighborhood__Gilbert  Neighborhood__IDOTRR  \
0                      0                      0                     0   
1                      0                      0                     0   
2                      0                      0                     0   
3                      0                      0                     0   
4                      0                      0                     0   

   Neighborhood__MeadowV  Neighborhood__Mitchel  Neighborhood__NAmes  \
0                      0                      0                    0   
1                      0                      0                    0   
2                      0                      0                    0   
3                      0                      0                    0   
4                      0                      0                    0   

   Neighborhood__NPkVill  Neighborhood__NWAmes  Neighborhood__NoRidge  \
0                      0                     0                      0   
1                      0                     0                      0   
2                      0                     0                      0   
3                      0                     0                      0   
4                      0                     0                      1   

   Neighborhood__NridgHt  Neighborhood__OldTown  Neighborhood__SWISU  \
0                      0                      0                    0   
1                      0                      0                    0   
2                      0                      0                    0   
3                      0                      0                    0   
4                      0                      0                    0   

   Neighborhood__Sawyer  Neighborhood__SawyerW  Neighborhood__Somerst  \
0                     0                      0                      0   
1                     0                      0                      0   
2                     0                      0                      0   
3                     0                      0                      0   
4                     0                      0                      0   

   Neighborhood__StoneBr  Neighborhood__Timber  Neighborhood__Ve

In [24]:
housing_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 75 columns):
 #   Column                 Non-Null Count  Dtype
---  ------                 --------------  -----
 0   LotArea                1460 non-null   int64
 1   TotalBsmtSF            1460 non-null   int64
 2   CentralAir             1460 non-null   int64
 3   1stFlrSF               1460 non-null   int64
 4   2ndFlrSF               1460 non-null   int64
 5   BedroomAbvGr           1460 non-null   int64
 6   KitchenAbvGr           1460 non-null   int64
 7   GarageArea             1460 non-null   int64
 8   PoolArea               1460 non-null   int64
 9   1stFlrSF               1460 non-null   int64
 10  2ndFlrSF               1460 non-null   int64
 11  SalePrice              1460 non-null   int64
 12  Neighborhood__Blueste  1460 non-null   int64
 13  Neighborhood__BrDale   1460 non-null   int64
 14  Neighborhood__BrkSide  1460 non-null   int64
 15  Neighborhood__ClearCr  1460 non-null  